packages/caver-wallet/src/keyringContainer.js

```diff
@@ -0,0 +1,308 @@
+
+        let keyring
+
+        if (_.isString(key)) keyring = Keyring.createWithSingleKey(address, key)
+
+        if (_.isArray(key)) {
+            if (key.length === 0) throw new Error(`Insufficient private key information: Empty array`)
+            if (_.isArray(key[0])) {
+                keyring = Keyring.createWithRoleBasedKey(address, key)
+            } else {
+                keyring = Keyring.createWithMultipleKey(address, key)
+            }
+        }
+
+        if (!(keyring instanceof Keyring)) throw new Error(`Unsupported type : ${typeof key}`)
+
+        return this.add(keyring)
+    }
+
+    /**
+     * updates the keyring inside the keyringContainer.
+     * Query the keyring to be updated from keyringContainer with the keyring's address,
+     * and an error occurs when the keyring is not found in the keyringContainer.
+     *
+     * @param {Keyring} keyring The keyring with new key.
+     * @return {Keyring}
+     */
+    updateKeyring(keyring) {
+        const founded = this._addressKeyringMap.get(keyring.address.toLowerCase())
+        if (founded === undefined) throw new Error(`Failed to find keyring to update`)
+
+        founded.key = keyring.copy().key
+        return founded
+    }
+
+    /**
+     * Get the keyring in container corresponding to the address
+     *
+     * @param {string} address The address of keyring to query.
+     * @return {Keyring}
+     */
+    getKeyring(address) {
+        if (!utils.isAddress(address))
+            throw new Error(
+                `Invalid address ${address}. To get keyring from wallet, you need to pass a valid address string as a parameter.`
+            )
+
+        const founded = this._addressKeyringMap.get(address.toLowerCase())
+
+        return founded
+    }
+
+    /**
+     * adds a keyring to the keyringContainer.
+     *
+     * @param {Keyring} keyring A keyring instance to add to keyringContainer.
+     * @return {Keyring}
+     */
+    add(keyring) {
+        if (this._addressKeyringMap.get(keyring.address.toLowerCase()) !== undefined)
+            throw new Error(`Duplicate Account ${keyring.address}`)
+
+        const keyringToAdd = keyring.copy()
+
+        this._length++
+        this._addressKeyringMap.set(keyringToAdd.address.toLowerCase(), keyringToAdd)
+
+        return keyringToAdd
+    }
+
+    /**
+     * deletes the keyring that associates with the given address from keyringContainer.
+     *
+     * @param {string} address An address of the keyring to be deleted in keyringContainer.
+     * @return {boolean}
+     */
+    remove(address) {
+        let keyringToRemove
+        if (utils.isAddress(address)) {
+            keyringToRemove = this.getKeyring(address)
+        } else {
+            throw new Error(`To remove keyring, parameter should be address string.`)
+        }
+
+        if (keyringToRemove === undefined) return false
+
+        keyringToRemove.key = null
```

Comment:
Is this statement necessary?

Reply:
There is possibility that the keyring object remains on memory, so this is to make sure key information is cleaned when user want to delete keyring.

Reply:
Could you please add a comment to describe the needs of this statement?

Reply:
I added comment for this :)